### ISSUE TITLE
Branch explanation

### ISSUE THREAD

**Issue by Swartz27** [Swartz27/xray-oxygen-rendering#2]
As I'm sure many now, I like rendering, a lot. So, I am re-writing the entire rendering system from scratch.  The reason for this is because not only are the renders an inefficient mess, but I'm also having a very difficult time understanding GSC's code style (whereas when I look at pretty much any other D3D11+ renderer it always makes sense and is nowhere near as messy). Until it is finished, I am working on cleaning up the existing system, especially stupid shader mistakes and bugs, and some new features too here and there.

**Current progress on existing renderer (R4):**

- Completed all to-do's left by dev's in the shaders
- Fixed a wide, and somewhat hilarious, amount of shader errors
- Fixed luminance calculation in Light_Render_Direct_ComputeXFS.cpp.....
- Variety of shader optimizations. It's pretty obvious that they ran out of time and had to learn D3D11 on the fly when it first came out, so a lot of these things aren't really their fault IMO
- Support for D3D11 features levels 11_0 - 12_1 (note: this is not a true DirectX12 renderer at all, I'm not sure how I should explain this one)
**WIP:**
- Gamma correction (done, but need to test out better solutions)
- Better tone mapping and bloom (50%, but only because I can't make up my mind)
- Voxel Ambient Occlusion (~75% done, works, but has some bad bugs and since I tested it as a pixel shader it's obviously very slow. Moving it to a geometry shader asap
- Voxel specular reflections (0%, but I already found a great way to do it. I may restrict this to only certain things as I imagine it could tank performance)

I am hoping to finish all this stuff asap, or, at least stabalize what is finished and commit it very soon.

**Plans for new "Uber-Renderer":**
- Support for D3D10.1 - D3D12.1
- New way of storing textures in memory. May not even work, but we'll see
- Both "classic" and "PBR" graphical paths
- Clustered Deffered Rendering
- Bastardized version of FX11 to accomplish the specific things I need only and nothing else
- Wide variety of modern rendering techniques (Temporal AA that doesn't suck, SVOGI, GBAA, MXAO, 
- Volumetric fog with lighting/shadowing, 3D-noise based clouds, Procedural sky (HoreSky I think it's called), new occlusion culling techniques, etc, etc,...
- All shading done inside one mega-shader, and no, this would not tank performance in the slightest, I was taught how to do this
- Other stuff too...

I am interested in working on this alone both because I'm a jerk and also because I want to make sure I learn everything properly without any time constraints. 

I am open to feature suggestions except for the following: I am not interest in emulating a camera, I prefer things looking as realistic as possible. This means no Depth of Field, no lens dirt, no filmic tonemappers (I've written my own more "natural" tonemapper which I like so far, but it needs more work), no film grain, etc.

tl;dr: I'm a dumb giant nerd.

BTW, I basically severed contact with everything related to "Radium". I don't even know if anyone bothered leaking the turd of a mod. Anyway, I also am planning on getting all the assets together that aren't crap and putting them in one big upload sometime soon. It will not be a functioning mod, it will be un-released features/files and various code snippets and notes.

**Comment by ForserX** [comment]
One request: 
* Synchronize stable developments with the main repo. :)

**Comment by Swartz27** [comment]
@ForserX Of course :)

**Comment by ForserX** [comment]
I'm added FX11 src and resseting this for oxygen build system.

**Comment by ForserX** [comment]
P.S. You can use oxy-commit-style? 

* TAG: Msg

**Comment by Swartz27** [comment]
@ForserX Yes, I will. Honestly I know very little about Git, and I tend not to follow conventions, but I'll read the documents you guys have to make sure I keep it compliant.

**Comment by Swartz27** [comment]
Open-ended question to all: Should I continue to essentially waste time working on the current renderer or would the smarter move be to devote the time I would normally spend on that and create the Uber Renderer now instead? I'm no expert coder, but I've seen so many damn DX11 code sources at this point that I can easily do it now. DX12 I'm going to have to learn, but the bits I've read so far don't seem as complicated as I had thought they would be.
Oh and I'm going to start popping into the Discord channel now and then, especially if I have any C++ questions.

**Comment by Swartz27** [comment]
Made up my mind: the only thing more I want to do for the existing renderer is simply to thoroughly test my uncommitted changes to make sure it all works properly, and I suppose I may as well finish the Voxel AO since that was one of the requests you guys have. As soon as that's done I'm beginning the new renderer, so probably within 1-2 days.

**Comment by ForserX** [comment]
Fine. I'm wait your PR.

**Comment by Xottab-DUTY** [comment]
I think it's better to improve existing renderer, than creating a new one from scratch.
Uber Renderer can be created from existing rendering system. Just do it step by step

**Comment by Xottab-DUTY** [comment]
Btw, DOF is some sort of realistic thing, not just camera emulation

**Comment by Swartz27** [comment]
@Xottab-DUTY I would improve the exisiting one but I honestly don't see the point, it honestly seems like it would actually be EASIER to re-write the renderer from scratch. I don't think GSC could have written worse rendering code if they had gone out of their way to do so.

As for DOF, can you explain? I'm not being a smart-ass, I'm just curious as I can't really think of any case where I find this to be true. However, there is a shader for Reshade that is called "DepthHaze" that I do consider realistic, as it is not DOF in the traditional sense. You can find the shader here: https://github.com/crosire/reshade-shaders/blob/master/Shaders/DepthHaze.fx
